sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_internal/async_polling.py

```diff
@@ -0,0 +1,84 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import base64
+from typing import TYPE_CHECKING
+
+from azure.core.polling.async_base_polling import AsyncLROBasePolling
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+
+class KeyVaultAsyncBackupClientPollingMethod(AsyncLROBasePolling):
+    def initialize(self, client, initial_response, deserialization_callback):
+        """Set the initial status of this LRO.
+
+        :param initial_response: The initial pipeline response of the poller, or a URL continuation token
+        :raises: HttpResponseError if initial status is incorrect LRO state
+        """
+        if isinstance(initial_response, str):
+            self._client = client
+            self._deserialization_callback = deserialization_callback
+            self._operation = self._lro_algorithms[0]
+            self._operation._async_url = initial_response  # pylint: disable=protected-access
+            self._status = "InProgress" # assume the operation is ongoing for now, so the actual status gets polled
+
+        else:
+            super(KeyVaultAsyncBackupClientPollingMethod, self).initialize(
+                client, initial_response, deserialization_callback
+            )
+
+    def get_continuation_token(self):
+        # type() -> str
+        return base64.b64encode(self._operation.get_polling_url().encode()).decode("ascii")
+
+    @classmethod
+    def from_continuation_token(cls, continuation_token, **kwargs):
+        """continuation_token is expected to be the URL of a pending operation"""
+        # type(str, Any) -> Tuple
+        try:
+            client = kwargs["client"]
+        except KeyError:
+            raise ValueError("Need kwarg 'client' to be recreated from continuation_token")
+
+        try:
+            deserialization_callback = kwargs["deserialization_callback"]
+        except KeyError:
+            raise ValueError("Need kwarg 'deserialization_callback' to be recreated from continuation_token")
+
+        continuation_url = base64.b64decode(continuation_token.encode()).decode("ascii")
+        return client, continuation_url, deserialization_callback
+
+    def _parse_resource(self, pipeline_response):
```

Comment:
Overriding a private method is risky because its signature, expected behavior, usage, and existence can change between versions. Is there a public caller we could override instead?

Reply:
That's a good point -- I can look for one